typescript-view-swapping/main.ts

```diff
@@ -0,0 +1,33 @@
+const $TabContainer = document.querySelector('.tab-container');
+const $AllTab = document.querySelectorAll('.tab');
+const $AllView = document.querySelectorAll('.view');
+
+if (!$TabContainer) throw new Error('The $TabContainer query did not work.');
+
+function handleClick(event: Event): void {
+  const $eventTarget = event.target as HTMLDivElement;
+
+  if ($eventTarget.matches('.tab') !== true) {
+    throw new Error('The $eventTarget is not selecting a tab');
+  }
```

Comment:
To make this even more brief and clear, we can just check for `not` the `$eventTarget.matches('.tab')`

`if (!$eventTarget.matches('.tab') throw new Error('$eventTarget is not selecting a tab`)`

This means that if the return of the `matches` method on the `$eventTarget` object comes back as any falsy value, it will throw the error.